[Bug]: Type mismatch warning

### Summary of Problem

**Description:**

Compiler warning
```bash
comm-ofi.c: In function ‘isAtomicValid’:
comm-ofi.c:8000:36: warning: comparison between ‘enum fi_datatype’ and ‘enum <anonymous>’ [-Wenum-compare]
 8000 |     for (enum fi_datatype t = 0; t < FI_DATATYPE_LAST; t++) {
```

**Is this issue currently blocking your progress?**
No

### Steps to Reproduce

Build Chapel with  `-Wenum-compare` flag.

**Source Code:**

 See https://github.com/chapel-lang/chapel/blob/main/runtime/src/comm/ofi/comm-ofi.c#L8000


**Compile command:**
N/A

**Execution command:**
N/A

**Associated Future Test(s):**
N/A

### Configuration Information
N/A

--- a/runtime/src/comm/ofi/comm-ofi.c
+++ b/runtime/src/comm/ofi/comm-ofi.c
@@ -7993,7 +7993,7 @@ int isAtomicValid(enum fi_datatype ofiType) {
   static int validByType[FI_DATATYPE_LAST];
 
   if (!inited) {
-    for (enum fi_datatype t = 0; t < FI_DATATYPE_LAST; t++) {
+    for (enum fi_datatype t = 0; t < (enum fi_datatype)FI_DATATYPE_LAST; t++) {
       validByType[t]  = computeAtomicValid(t);
     }
     inited = true;
@@ -8009,11 +8009,11 @@ int isAtomicValid(enum fi_datatype ofiType) {
       struct fid_ep* ep = tciTab[0].txCtx; // assume same answer for all
                                            // endpoints
 
-      for (enum fi_datatype t = 0; t < FI_DATATYPE_LAST; t++) {
+      for (enum fi_datatype t = 0; t < (enum fi_datatype)FI_DATATYPE_LAST; t++) {
         offset = 0;
         offset += snprintf(buf + offset, sizeof(buf) - offset, "%s: ",
                       fi_tostr(&t, FI_TYPE_ATOMIC_TYPE));
-        for (enum fi_op op = 0; op < FI_ATOMIC_OP_LAST; op++) {
+        for (enum fi_op op = 0; op < (enum fi_op)FI_ATOMIC_OP_LAST; op++) {
           size_t count; // needed by macros below
           int valid = my_valid(t, op);
           int fetch = my_fetch_valid(t, op);
@@ -8038,7 +8038,7 @@ int isAtomicValid(enum fi_datatype ofiType) {
         }
         DBG_PRINTF(DBG_CFG_AMO, "%s", buf);
       }
-      for (enum fi_datatype t = 0; t < FI_DATATYPE_LAST; t++) {
+      for (enum fi_datatype t = 0; t < (enum fi_datatype)FI_DATATYPE_LAST; t++) {
         DBG_PRINTF(DBG_CFG_AMO, "%s: %s", fi_tostr(&t, FI_TYPE_ATOMIC_TYPE),
                    validByType[t] ? "valid" : "invalid");
       }

